url.go

```diff
@@ -211,6 +226,61 @@ func parseReference(url string) (*Reference, error) {
 		}
 		break
 	}
+	if r.User != "" {
+		if !names.IsValidUser(r.User) {
+			return nil, fmt.Errorf("entity URL has invalid user name: %q", url)
+		}
+	}
+	if !IsValidName(r.Name) {
+		return nil, fmt.Errorf("entity URL has invalid entity name: %q", url)
+	}
+	return &r, nil
+}
+
+func parseV2Reference(url *gourl.URL) (*Reference, error) {
+	var r Reference
+	r.Schema = "cs"
+	parts := strings.Split(strings.Trim(url.Path, "/"), "/")
+	if len(parts) == 0 {
+		return nil, fmt.Errorf("entity URL malformed, no parts: %q", url)
```

Comment:
This can never happen (strings.Split always returns at least one element in the slice)